teek/_widgets/treeview.py

```diff
@@ -0,0 +1,469 @@
+                "this column hasn't been added to a treeview yet")
+
+    def _item_handler(self, rettype, option, *args):
+        self._check_added()
+        return self._treeview._call(rettype, self._treeview, 'item', self,
+                                    '-' + option, *args)
+
+    @make_thread_safe
+    def _assign(self, treeview):
+        # Assign row to given treeview
+        self._treeview = treeview
+
+        for name, value in self._creation_opts.items():
+            self.config[name] = value
+
+    @make_thread_safe
+    def to_tcl(self):
+        # Represented by name
+        return self._name
+
+    @property
+    def selected(self):
+        """
+        Get current selection state
+        """
+        self._check_added()
+        return self._name in self._treeview._call([str], self._treeview,
+                                                  'selection')
+
+    @make_thread_safe
+    def select(self):
+        """
+        Add row to current selection
+        """
+        self._check_added()
+        self._treeview._call(None, self._treeview, 'selection', 'add', self)
+
+    @make_thread_safe
+    def deselect(self):
+        """
+        Remove row from current selection
+        """
+        self._check_added()
+        self._treeview._call(None, self._treeview, 'selection', 'remove', self)
+
+
+class TreeviewColumnList(MutableSequence):
+    """
+    List containing all columns of a treeview
+    """
+
+    def __init__(self, treeview, init_columns=[]):
+        super().__init__()
+
+        # Default column 0
+        first_column = TreeviewColumn('#0')
+        first_column._assign(treeview)
+
+        self._data = [first_column]
+        self._treeview = treeview
+
+        for column in init_columns:
+            self.append(column)
+
+    def __setitem__(self, index, column):
+        if index == 0:
+            raise KeyError('cannot set column #0')
+
+        if not isinstance(column, TreeviewColumn):
+            column = TreeviewColumn(text=column)
+
+        self._data[index] = column
+        self._update()
+
+    def __delitem__(self, index):
+        if index == 0:
+            raise KeyError('cannot delete column #0')
+
+        del self._data[index]
+        self._update()
+
+    def __getitem__(self, index):
+        return self._data[index]
+
+    def __len__(self):
+        # Length includes column 0
+        return len(self._data)
+
+    def _update(self):
+        """
+        Update columns of widget and rebuild column objects
+        """
+        self._treeview._call(None, self._treeview, 'configure', '-columns',
+                             self._data[1:])
+
+        for column in self._data:
+            column._assign(self._treeview)
+
+    def insert(self, index, column):
+        """
+        Insert column or column text converted to a column at given index.
+        Note that the column at index 0 cannot be replaced!
+        """
+        if index == 0:
```

Comment:
This check isn't very good, because e.g. `-100` will likely insert at the beginning

```
>>> asdf = [1, 2, 3]
>>> asdf.insert(-100, 'lol')
>>> asdf
['lol', 1, 2, 3]
>>> 
```

something like this instead (I didn't try to run it):

```
if index < -len(self):
    index = -len(self)
if index < 0:
    index += len(self)
if index == 0:
    # this runs when inserting to the beginning
```

or even better, let the list handle all this, and check `if self.data[0] == column` after the insert.